src/shared/components/query/GenesetsVolcanoSelector.tsx

```diff
@@ -0,0 +1,313 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import LabeledCheckbox from "../labeledCheckbox/LabeledCheckbox";
+import * as styles_any from './styles/styles.module.scss';
+import {action, ObservableMap, expr, toJS, computed, observable, autorun} from "mobx";
+import {observer, Observer} from "mobx-react";
+import EnhancedReactTable from "../enhancedReactTable/EnhancedReactTable";
+import {Geneset} from "../../api/generated/CBioPortalAPIInternal";
+import {IColumnFormatterData} from "../enhancedReactTable/IColumnFormatter";
+import {IColumnDefMap} from "../enhancedReactTable/IEnhancedReactTableProps";
+import {Td} from 'reactable';
+import {toPrecision} from "../../lib/FormatUtils";
+import { getHierarchyData } from "shared/lib/StoreUtils";
+import ReactSelect from "react-select";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import { VictoryChart, VictoryScatter, VictoryTheme, VictorySelectionContainer,
+    VictoryAxis, VictoryLabel, VictoryLine } from 'victory';
+
+const styles = styles_any as {
+    GenesetsVolcanoSelectorWindow: string,
+    selectButton: string,
+    selectionColumnHeader: string,
+    selectionColumnCell: string,
+};
+
+class GenesetsVolcanoTable extends EnhancedReactTable<Geneset> {}
+
+export interface GenesetsVolcanoSelectorProps
+{
+    gsvaProfile: string;
+    sampleListId: string|undefined;
+    onSelect: (map_genesets_selected:ObservableMap<boolean>) => void;
+}
+
+@observer
+export default class GenesetsVolcanoSelector extends React.Component<GenesetsVolcanoSelectorProps, {plotData:{x: number, y: number, fill: string}[]}>
+{
+    @observable tableData: Geneset[] = [];
+    @observable allTableData: Geneset[] = [];
+    maxY = 1;
+    plotData = autorun(() => {
```

Comment:
elsewhere in the we've avoided using React state, instead just using observable properties and computed getters.  Is there a reason not to just make this @computed getter?